metasynth/distribution/base.py

```diff
@@ -42,17 +43,19 @@ def fit(cls, series: Sequence, *args, **kwargs) -> BaseDistribution:
 
     @staticmethod
     def _to_series(values: Sequence):
-        if isinstance(values, pandas.Series):
-            series = values.dropna()
+        if isinstance(values, pl.Series):
+            series = values.drop_nulls()
+        elif isinstance(values, pd.Series):
+            series = pl.Series(values).drop_nulls()  # pylint: disable=assignment-from-no-return
```

Comment:
So this is where the conversion happens. This was extensively tested for different data types yeah?

Reply:
It should work for most at least arrays and pandas DataFrames, probably lists as well.